src/EventStore.Core/Cluster/EventStoreClusterClient.cs

```diff
@@ -15,10 +17,31 @@ public partial class EventStoreClusterClient : IDisposable {
 		private readonly IPublisher _bus;
 		internal bool Disposed { get; private set; }
 
-		public EventStoreClusterClient(Uri address, IPublisher bus,
+		public EventStoreClusterClient(Uri address, IPublisher bus, string tlsTargetHost = null,
 			Func<HttpMessageHandler> httpMessageHandlerFactory = null) {
+			if(tlsTargetHost == null && address.Scheme == Uri.UriSchemeHttps)
+				throw new Exception("Address scheme is https but TLS target host not specified.");
+
+			HttpMessageHandler httpMessageHandler = httpMessageHandlerFactory?.Invoke() ?? new HttpClientHandler {
+				ServerCertificateCustomValidationCallback = (message, certificate, chain, errors) => {
```

Comment:
Is there any reason why we are performing our own certificate validation here?

Reply:
There are two main reasons:
- To be able to validate the `TlsTargetHost` - to align the behaviour with how TLS currently works inside eventstore.
- We can log specific error messages when something does not match to make the user aware about the problem, otherwise it fails silently.